api/src/main/java/org/openmrs/api/db/hibernate/HibernateConceptDAO.java

```diff
@@ -1016,54 +1016,29 @@ public void remove() {
 		}
 		
 	}
-	
+
 	/**
 	 * @see org.openmrs.api.db.ConceptDAO#getConceptsByMapping(String, String, boolean)
 	 */
 	@Override
 	@SuppressWarnings("unchecked")
 	public List<Concept> getConceptsByMapping(String code, String sourceName, boolean includeRetired) {
-		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(ConceptMap.class);
-		
-		// make this criteria return a list of concepts
+		Criteria criteria = createSearchConceptMapCriteria(code, sourceName, includeRetired);
```

Comment:
I'd make use of cache here as well with:
```
List<Integer> ids = Context.getConceptService().getConceptIdsByMapping(code, sourceName, includeRetired); 
session.createCriteria(Concept.class).add(Restrictions.in("id",ids)).list(); 
if (includeRetired) {
			// sort retired concepts to the end of the list
			criteria.addOrder(Order.asc("retired"));
}
```

Reply:
I tried this out, and it was a lot slower (2x) than the way the service method is implemented.  I also am not a fan of services calling DAOs which then call different services.  I actually think we should leave this dao method implementation alone but Deprecate the method, as it is no longer used by the Service layer at all.

Reply:
Interesting. Fair point to not call services in DAOs.

Reply:
I assume this is due to the fact that by iterating over the concept ids and calling getConcept(id), we also leverage hibernate's use of ehcache in a more efficient manner than by running an HQL query with an "in" clause on ids.

Reply:
Am in favour of leaving this dao method implementation alone.